pkg/notify/notifier/sms/interface.go

```diff
@@ -29,16 +29,21 @@ func Register(name string, p ProviderFactory) {
 	availableFactoryFuncs[name] = p
 }
 
-func GetProviderFunc(name string) (ProviderFactory, bool) {
-	p, ok := availableFactoryFuncs[name]
-	return p, ok
-}
-
-func GetFirstAvailableProviderFunc() (ProviderFactory, error) {
-	for _, p := range availableFactoryFuncs {
-		if p != nil {
-			return p, nil
+func GetProviderFunc(name string) (ProviderFactory, error) {
+	if name != "" {
+		// check whether the default provider is registered
+		p, ok := availableFactoryFuncs[name]
+		if !ok {
+			return nil, errors.New("the given default sms provider not registered")
+		}
+		return p, nil
+	} else {
+		// use the first available provider func if the default provider not given
+		for _, p := range availableFactoryFuncs {
+			if p != nil {
+				return p, nil
```

Comment:
do not return loop variable directly, assign it to another variable and return that

Reply:
fixed